public/app/plugins/datasource/influxdb/components/AnnotationQueryEditor.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { PureComponent } from 'react';
+import coreModule from 'app/core/core_module';
+import kbn from 'app/core/utils/kbn';
+
+import { DataSourceApi, AnnotationEvent, DataQueryResponse, LoadingState } from '@grafana/data';
+import { AnnotationFieldMapper, AnnotationsFromFrameOptions } from '@grafana/runtime';
+import { Spinner, Icon } from '@grafana/ui';
+
+import { InfluxQuery, InfluxAnnotation } from '../types';
+import { FluxQueryEditor } from './FluxQueryEditor';
+import { getDashboardSrv } from 'app/features/dashboard/services/DashboardSrv';
+import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';
+import { cx, css } from 'emotion';
+
+interface Props {
+  datasource: DataSourceApi;
+  annotation: InfluxAnnotation;
+  change: (query: InfluxAnnotation) => void;
+}
+
+interface State {
+  running?: boolean;
+  rsp?: DataQueryResponse;
+  events?: AnnotationEvent[];
+}
+
+export class AnnotationQueryEditor extends PureComponent<Props, State> {
+  state = {} as State;
+
+  componentDidMount() {
+    this.runQuery();
+  }
+
+  componentDidUpdate(oldProps: Props) {
+    if (this.props.annotation !== oldProps.annotation) {
+      this.runQuery();
+    }
+  }
+
+  runQuery = async () => {
```

Comment:
should this really be in the editor, can't we move this to Grafana app/annotation editor view?  (I know it's in angular and a lot of work to do there to migrate and rethink that)

Reply:
should it be?  NO!

ideally this would be the default behavior if you do not specify anything... but given our setup that is a much larger project.  My plan here is to make things in runtime as generic as possible so that anything that extends DataSourceWithBackend can just use them, and when we put the effort towards annotations in general we can reuse this component/path